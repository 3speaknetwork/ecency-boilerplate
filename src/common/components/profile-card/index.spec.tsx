import React from "react";

import ProfileCard from "./index";
import renderer from "react-test-renderer";

jest.mock("../../constants/defaults.json", () => ({
  imageServer: "https://images.esteem.app",
}));

const account = {
  active: {
    account_auths: [],
    key_auths: [["STM71z4rmzGHdp7pmePZyS1G2GrbuBqg9PPHRgiK6uYqjzUTRNCRR", 1]],
    weight_threshold: 1,
  },
  balance: "2.598 HIVE",
  can_vote: true,
  comment_count: 0,
  created: "2018-01-29T20:10:06",
  curation_rewards: 53,
  delegated_vesting_shares: "0.000000 VESTS",
  downvote_manabar: { current_mana: "16874750664", last_update_time: 1589888058 },
  guest_bloggers: [],
  id: 689709,
  json_metadata:
    '{"profile":{"name":"Talha B.","about":"Founder @runkod, Software Developer @esteemapp, Lifelong Learner","cover_image":"https://img.esteem.app/rwd380.jpg","profile_image":"https://img.esteem.app/821e0q.jpg","website":"https://github.com/talhasch","location":"Istanbul"},"escrow":{"fees":{"STEEM":"1.000","SBD":"3.000"},"terms":"lorem ipsum dolor sit amet"}}',
  last_account_recovery: "1970-01-01T00:00:00",
  last_account_update: "2020-05-25T17:00:24",
  last_owner_update: "1970-01-01T00:00:00",
  last_post: "2020-04-28T07:05:27",
  last_root_post: "2020-04-28T07:04:18",
  last_vote_time: "2020-03-31T07:11:57",
  lifetime_vote_count: 0,
  market_history: [],
  memo_key: "STM7rioGL7NopT2Zo446hnzAiHLp5sJ1gS55xu3NikmxfQnEznsue",
  mined: false,
  name: "talhasch",
  next_vesting_withdrawal: "1969-12-31T23:59:59",
  other_history: [],
  owner: {
    account_auths: [],
    key_auths: [["STM7F7zfd6ieangxz6uxQkYifUS5H841x5E41SYZaVc9F9cGPJ9jN", 1]],
    weight_threshold: 1,
  },
  pending_claimed_accounts: 0,
  post_bandwidth: 0,
  post_count: 212,
  post_history: [],
  posting: {
    account_auths: [
      ["busy.app", 1],
      ["drugwars.app", 1],
      ["esteem-app", 1],
      ["esteem.app", 1],
      ["esteemapp", 1],
    ],
    key_auths: [["STM6uvU7j624wCZNa2pcXvkqmbnNC1cgraDrVG3pFRhXvj9LYS7Xp", 1]],
    weight_threshold: 1,
  },
  posting_json_metadata:
    '{"profile":{"profile_image":"https://scontent-lht6-1.xx.fbcdn.net/v/t1.0-1/p320x320/12241236_10153810817671310_7266508310252207215_n.jpg?oh=1fa175937c9c76910e106f443264a2a1&oe=5B491036","name":"Talha B.","location":"Istanbul","website":"https://github.com/talhasch","cover_image":"","about":""},"escrow":{"fees":{"STEEM":"1.000","SBD":"3.000"},"terms":"lorem ipsum dolor sit amet"}}',
  posting_rewards: 1859,
  proxied_vsf_votes: [0, 0, 0, 0],
  proxy: "",
  received_vesting_shares: "0.000000 VESTS",
  recovery_account: "steem",
  reputation: "33082349040",
  reset_account: "null",
  reward_sbd_balance: "0.000 HBD",
  reward_steem_balance: "0.000 HIVE",
  reward_vesting_balance: "0.000000 VESTS",
  reward_vesting_steem: "0.000 HIVE",
  savings_balance: "0.000 HIVE",
  savings_sbd_balance: "0.000 HBD",
  savings_sbd_last_interest_payment: "2019-01-09T15:17:45",
  savings_sbd_seconds: "0",
  savings_sbd_seconds_last_update: "2019-01-09T15:17:45",
  savings_withdraw_requests: 0,
  sbd_balance: "0.006 HBD",
  sbd_last_interest_payment: "2019-12-19T11:16:00",
  sbd_seconds: "0",
  sbd_seconds_last_update: "2020-04-02T08:48:39",
  tags_usage: [],
  to_withdraw: "87017000000",
  transfer_history: [],
  vesting_balance: "0.000 HIVE",
  vesting_shares: "67499.002656 VESTS",
  vesting_withdraw_rate: "0.000000 VESTS",
  vote_history: [],
  voting_manabar: { current_mana: "67499002656", last_update_time: 1589888058 },
  voting_power: 0,
  withdraw_routes: 1,
  withdrawn: "87017000000",
  witness_votes: [
    "aggroed",
    "anyx",
    "ausbitbank",
    "blocktrades",
    "emrebeyler",
    "good-karma",
    "gtg",
    "roelandp",
    "someguy123",
    "steempeak",
    "themarkymark",
    "therealwolf",
    "yabapmatt",
  ],
  witnesses_voted_for: 13,
  profile: {
    name: "Talha B.",
    about: "Founder @runkod, Software Developer @esteemapp, Lifelong Learner",
    cover_image: "https://img.esteem.app/rwd380.jpg",
    profile_image: "https://img.esteem.app/821e0q.jpg",
    website: "https://github.com/talhasch",
    location: "Istanbul",
  },
  follow_stats: { account: "talhasch", follower_count: 210, following_count: 29 },
};

it("(1) Default render", () => {
  const props = {
    account,
  };

  //@ts-ignore
  const component = renderer.create(<ProfileCard {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
