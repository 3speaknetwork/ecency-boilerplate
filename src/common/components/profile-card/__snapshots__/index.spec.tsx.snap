// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Default render 1`] = `
<div
  className="profile-card"
>
  <div
    className="profile-avatar"
  >
    <span
      className="user-avatar xLarge"
      onClick={[Function]}
      role="none"
      style={
        Object {
          "backgroundImage": "url('https://images.esteem.app/u/talhasch/avatar/large')",
        }
      }
    />
    <div
      className="reputation"
    >
      38
    </div>
  </div>
  <div
    className="username"
  >
    talhasch
  </div>
  <div
    className="vpower-line"
  >
    <div
      className="vpower-line-inner"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="vpower-percentage"
  >
    <span
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      100.00
    </span>
  </div>
  <div
    className="basic-info"
  >
    <div
      className="full-name"
    >
      Talha B.
    </div>
    <div
      className="about"
    >
      Founder @runkod, Software Developer @esteemapp, Lifelong Learner
    </div>
  </div>
  <div
    className="stats"
  >
    <div
      className="stat"
    >
      <span
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <svg
          viewBox="0 0 24 24"
        >
          <path
            d="M7,5H21V7H7V5M7,13V11H21V13H7M4,4.5A1.5,1.5 0 0,1 5.5,6A1.5,1.5 0 0,1 4,7.5A1.5,1.5 0 0,1 2.5,6A1.5,1.5 0 0,1 4,4.5M4,10.5A1.5,1.5 0 0,1 5.5,12A1.5,1.5 0 0,1 4,13.5A1.5,1.5 0 0,1 2.5,12A1.5,1.5 0 0,1 4,10.5M7,19V17H21V19H7M4,16.5A1.5,1.5 0 0,1 5.5,18A1.5,1.5 0 0,1 4,19.5A1.5,1.5 0 0,1 2.5,18A1.5,1.5 0 0,1 4,16.5Z"
            fill="currentColor"
          />
        </svg>
         
        212
      </span>
    </div>
    <div
      className="stat"
    >
      <span
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <svg
          viewBox="0 0 24 24"
        >
          <path
            d="M16 17V19H2V17S2 13 9 13 16 17 16 17M12.5 7.5A3.5 3.5 0 1 0 9 11A3.5 3.5 0 0 0 12.5 7.5M15.94 13A5.32 5.32 0 0 1 18 17V19H22V17S22 13.37 15.94 13M15 4A3.39 3.39 0 0 0 13.07 4.59A5 5 0 0 1 13.07 10.41A3.39 3.39 0 0 0 15 11A3.5 3.5 0 0 0 15 4Z"
            fill="currentColor"
          />
        </svg>
         
        210
      </span>
    </div>
    <div
      className="stat"
    >
      <span
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <svg
          viewBox="0 0 24 24"
        >
          <path
            d="M15,14C12.33,14 7,15.33 7,18V20H23V18C23,15.33 17.67,14 15,14M6,10V7H4V10H1V12H4V15H6V12H9V10M15,12A4,4 0 0,0 19,8A4,4 0 0,0 15,4A4,4 0 0,0 11,8A4,4 0 0,0 15,12Z"
            fill="currentColor"
          />
        </svg>
         
        29
      </span>
    </div>
  </div>
  <div
    className="extra-props"
  >
    <div
      className="prop"
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M21,3L3,10.53V11.5L9.84,14.16L12.5,21H13.46L21,3Z"
          fill="currentColor"
        />
      </svg>
       
      Istanbul
    </div>
    <div
      className="prop"
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M17.9,17.39C17.64,16.59 16.89,16 16,16H15V13A1,1 0 0,0 14,12H8V10H10A1,1 0 0,0 11,9V7H13A2,2 0 0,0 15,5V4.59C17.93,5.77 20,8.64 20,12C20,14.08 19.2,15.97 17.9,17.39M11,19.93C7.05,19.44 4,16.08 4,12C4,11.38 4.08,10.78 4.21,10.21L9,15V16A2,2 0 0,0 11,18M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          fill="currentColor"
        />
      </svg>
      <a
        className="website-link"
        href="https://github.com/talhasch"
        target="_external"
      >
        https://github.com/talhasch
      </a>
    </div>
    <div
      className="prop"
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M9,10H7V12H9V10M13,10H11V12H13V10M17,10H15V12H17V10M19,3H18V1H16V3H8V1H6V3H5C3.89,3 3,3.9 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5A2,2 0 0,0 19,3M19,19H5V8H19V19Z"
          fill="currentColor"
        />
      </svg>
       
      January 29, 2018
    </div>
    <div
      className="prop"
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M6.18,15.64A2.18,2.18 0 0,1 8.36,17.82C8.36,19 7.38,20 6.18,20C5,20 4,19 4,17.82A2.18,2.18 0 0,1 6.18,15.64M4,4.44A15.56,15.56 0 0,1 19.56,20H16.73A12.73,12.73 0 0,0 4,7.27V4.44M4,10.1A9.9,9.9 0 0,1 13.9,20H11.07A7.07,7.07 0 0,0 4,12.93V10.1Z"
          fill="currentColor"
        />
      </svg>
      <a
        href="https://ecency.com/@talhasch/rss"
        target="_external"
      >
        RSS feed
      </a>
    </div>
  </div>
</div>
`;
